model/project.go

```diff
@@ -372,6 +375,44 @@ type ProjectTask struct {
 	PatchOnly *bool `yaml:"patch_only,omitempty" bson:"patch_only,omitempty"`
 }
 
+type LoggerConfig struct {
+	Type         string `yaml:"type" bson:"type"`
+	SplunkServer string `yaml:"splunk_server,omitempty" bson:"splunk_server,omitempty"`
+	SplunkToken  string `yaml:"splunk_token,omitempty" bson:"splunk_token,omitempty"`
+}
+
+func (c *LoggerConfig) IsValid() error {
+	catcher := grip.NewBasicCatcher()
+	if !util.StringSliceContains(ValidLogSenders, c.Type) {
+		catcher.Add(errors.Errorf("%s is not a valid log sender", c.Type))
+	}
+	if c.Type == string(SplunkLogSender) && c.SplunkServer == "" {
+		catcher.Add(errors.New("Splunk logger requires a server URL"))
+	}
+	if c.Type == string(SplunkLogSender) && c.SplunkToken == "" {
+		catcher.Add(errors.New("Splunk logger requires a token"))
+	}
+
+	return catcher.Resolve()
+}
+
+type LogSender string
+
+const (
+	EvergreenLogSender LogSender = "evergreen"
+	FileLogSender                = "file"
+	LogkeeperLogSender           = "logkeeper"
+	SplunkLogSender              = "splunk"
+)
+
+var ValidLogSenders = []string{
+	string(EvergreenLogSender),
+	string(FileLogSender),
+	string(LogkeeperLogSender),
+	string(SplunkLogSender),
+	"",
```

Comment:
If the reason for the empty string being a valid log sender because we have a fallback, you could use the validate method to check for the empty string. That way `ValidLogSenders` contains only truly valid senders, and we set the empty string to the default somewhere before we validate.

Reply:
I feel like this approach is better since you can round trip the yaml without it being modified (I think)

Reply:
Round tripping the YAML will modify it in pretty substantial ways, so we can't really hold on to the original state. The main reason to not to have this in my view is that it's harder to validate log senders with this slice. It could lead to bugs where people expect the validation method to return invalid for the empty string.